### PR TITLE
download orbit

### PR DESCRIPTION
**Pull request by hfattahi**
Borrowing the fetchorbit script from our isce2 stack processor and slightly modifying so that a directory with all safe files can be given as input to loop over and download the orbits for.

### CODE REVIEW

File: src/compass/download_orbits.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+def fetch_orbit(safe_file, orbit_dir):
+    fileTS, satName, fileTSStart = FileToTimeStamp(safe_file)
+    print('Reference time: ', fileTS)
+    print('Satellite name: ', satName)
+    match = None
+    session = requests.Session()
+
+    for spec in orbitMap:
+        oType = spec[0]
```
Comment (vbrancat) [inline src/compass/download_orbits.py:131]:
```suggestion
        o_type = spec[0]
```

File: src/compass/download_orbits.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        success = False
+        match = None
+  
+        try:
+            r = session.get(url, verify=True, auth=credentials)
+            r.raise_for_status()
+            parser = MyHTMLParser(url)
+            parser.feed(r.text)
+            for resulturl, result in parser.fileList:
+                tbef, taft, mission = fileToRange(os.path.basename(result))
```
Comment (vbrancat) [inline src/compass/download_orbits.py:146]:
```suggestion
                tbef, taft, mission = file_to_range(os.path.basename(result))
```

File: src/compass/download_orbits.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+#Generic credentials to query and download orbit files
+credentials = ('gnssguest', 'gnssguest')
+
+
+def cmdLineParse():
+    '''
+    Command line parser.
+    '''
+
+    parser = argparse.ArgumentParser(description='Fetch Sentinel-1 orbits')
```
Comment (vbrancat) [inline src/compass/download_orbits.py:30]: May be clarify the logic with which the orbits are downloaded, i.e. look for POE first if POE is not found download the ROE?

**Review by vbrancat — COMMENTED** [review]
@hfattahi thanks a lot for these PR. My major comment is to try to uniform the coding style to pep8 when possible. Also while testing the script, I ran into an issue (solvable, I think by modifying the output path to which the orbits are allocated). Also, I think the script should go into the `utils` folder as it is a functionality more then a workflow. @LiangJYu and @yunjunz would for sure have an opinion on that.

**Comment by yunjunz** [comment]
> @hfattahi thanks a lot for these PR. My major comment is to try to uniform the coding style to pep8 when possible. Also while testing the script, I ran into an issue (solvable, I think by modifying the output path to which the orbits are allocated). Also, I think the script should go into the `utils` folder as it is a functionality more then a workflow. @LiangJYu and @yunjunz would for sure have an opinion on that.

Since this script could be potentially added to a command-line executable, I like it under `src/compass`.

For the naming, shall we make it consistent with other scripts, e.g. `s1_dload_orbit.py`?

**Comment by yunjunz** [comment]
For the convenience and flexibility on the user side, it will be great if we could support:

1. the input SLC path as safe file, in addition to the directory of it.
2. `--start` and `--end` date, without input safe file/dir, as what we have in `topsStack/dloadOrbit.py`.

**Comment by LiangJYu** [comment]
I think this code is a better fit in [s1-reader](https://github.com/opera-adt/s1-reader) given the substantial amount of duplicate functionality in [s1_orbit.py](https://github.com/opera-adt/s1-reader/blob/main/src/s1reader/s1_orbit.py). @hfattahi if you agree, I will resubmit PR in s1-reader.

**Comment by hfattahi** [comment]
> I think this code is a better fit in [s1-reader](https://github.com/opera-adt/s1-reader) given the substantial amount of duplicate functionality in [s1_orbit.py](https://github.com/opera-adt/s1-reader/blob/main/src/s1reader/s1_orbit.py). @hfattahi if you agree, I will resubmit PR in s1-reader.

That is fine. Also I came across this script as well https://github.com/scottstanie/sentineleof . It may does all we need

**Comment by vbrancat** [comment]
@hfattahi we have moved the orbit download functionality to a new PR in the `s1-reader`. If you don't mind, I would be inclined to close this PR and move the conversation to the `s1-reader` repo. Keep the branch just in case. What do you think?

**Comment by vbrancat** [comment]
Closing since functionality has been included in the s1-reader

**Pull request closed without merging**